src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -675,49 +851,199 @@ private static IEnumerable<double> LazySequence(double records, double start, do
+            int digits;
+
+            if (args.Length == 2 && args[1] is NumberValue numberDigs)
+            {
+                digits = (int)numberDigs.Value;
+            }
+            else if (args.Length == 2 && args[1] is DecimalValue decimalDigs)
+            {
+                digits = (int)decimalDigs.Value;
+            }
+            else
+            { 
+                return CommonErrors.UnreachableCodeError(irContext);
+            }
 
-            if (dg < -15 || dg > 15 || number < -1e20d || number > 1e20d)
+            if (args[0] is NumberValue num)
+            {
+                return RoundFloat(irContext, num, digits);
+            }
+            else if (args[0] is DecimalValue dec)
             {
-                return number;
+                return RoundDecimal(irContext, dec, digits);
             }
 
-            // Dividing by m, since multiplication was introducing floating point error
-            var m = Math.Pow(10d, dg);
-            var eps = 1 / (m * 1e12d); // used to manage rounding of 1.4499999999999999999996
+            return CommonErrors.UnreachableCodeError(irContext);
+        }
 
-            switch (rt)
+        internal static FormulaValue RoundFloat(IRContext irContext, NumberValue num, int dg, RoundType rt = RoundType.Default)
+        {
+            var number = num.Value;
+            var s = number < 0 ? -1d : 1d;
+            var n = number * s;
+
+            if (dg < -15 || dg > 15 || number < -1e20d || number > 1e20d)
+            {
+                return num;
+            }
+
+            // Dividing by m, since multiplication was introducing floating point error
+            var m = Math.Pow(10d, dg);
+            var eps = 1 / (m * 1e12d); // used to manage rounding of 1.4499999999999999999996
+
+            switch (rt)
+            {
+                case RoundType.Default:
+                    return new NumberValue(irContext, s * Math.Floor((n + (1 / (2 * m)) + eps) * m) / m);
+                case RoundType.Down:
+                    return new NumberValue(irContext, s * Math.Floor(n * m) / m);
+                case RoundType.Up:
+                    return new NumberValue(irContext, s * Math.Ceiling(n * m) / m);
+            }
+
+            return CommonErrors.UnreachableCodeError(irContext);
+        }
+
+        // The algorithm for Decimal is different from that of Float because with less range we are going out of our way to avoid overflow
+        // At the time of this writing, the version of .NET being targeted only supports two varieties of MidpointRounding
+        // In the future, some of this can be replaced with built in support in decimal.Round()
+        internal static FormulaValue RoundDecimal(IRContext irContext, DecimalValue dec, int digits, RoundType roundType = RoundType.Default)
+        {
+            // Decimal TODO: This should be moved out as a static member, but tests complain
```

Comment:
@MikeStall , I still need a solution to this problem.  The test suite still complains if I try to make this a static in some manner.

Reply:
Fixed, #1341

Reply:
Thank you!